library/std/src/lib.rs

```diff
@@ -137,3 +137,11 @@ macro_rules! println {
 macro_rules! eprintln {
     ($($x:tt)*) => { evaluate_print_args!($($x)*); };
 }
+
+#[macro_export]
+macro_rules! unreachable {
+    ($($arg:tt)*) => ({
+        panic!(concat!("internal error: entered unreachable code: ",
```

Comment:
How does this compare to the std version?

Reply:
The message is basically the same except for the following:
 - we still inclue ":" in the no arguments case.
 - the std one use format
 - we print extra quotes for the message